packages/babel-parser/src/plugins/flow.js

```diff
@@ -901,9 +936,41 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           );
         }
         this.expect(tt.ellipsis);
-        node.argument = this.flowParseType();
+        switch (this.state.type) {
+          case tt.comma:
+          case tt.semi:
+          case tt.braceR:
+          case tt.braceBarR: {
+            const isInexactToken = this.eat(tt.comma) || this.eat(tt.semi);
+
+            if (this.match(tt.braceR)) {
+              if (allowInexact) return true;
+              this.unexpected(
+                null,
+                "Explicit inexact syntax is only allowed inside inexact objects",
+              );
+            }
+
+            if (this.match(tt.braceBarR)) {
+              this.unexpected(
+                null,
+                "Explicit inexact syntax cannot appear inside an explicit exact object type",
+              );
+              return false;
```

Comment:
Those `retuen`s aren't needed, since `this.unexpected` already throws. If you want to be explicit, I think that it would be more correct to use `throw this.unexpected`.